pkg/build/builder/dockerutil.go

```diff
@@ -89,6 +120,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	return err
 }
 
+func getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error {
+	glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.")
+	timeout := time.After(time.Second * 2)
+
+waitLoop:
+	for {
```

Comment:
I believe you mean [this Until](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/wait/wait.go#L46). I can update it at once.